lib/media/segment_index.js

```diff
@@ -129,9 +129,24 @@ shaka.media.SegmentIndex.prototype.merge = function(references) {
       shaka.log.warning('Refusing to rewrite original references on update!');
       j++;
     } else {
-      // Drop the new reference if there's an old reference with the same
-      // time.
-      newReferences.push(r1);
+      // when period is changed, fitSegmentReference will
+      // expand last segment to the start of the next the period
+      // so, it is valid to have end time updated to the
+      // last segment reference in a period
+      if (r1.endTime != r2.endTime)
+      {
```

Comment:
Our style convention is to put curly braces on the previous line.  Please update.